docs/topics/data-analysis/data-analysis-connect-to-db.md

```diff
@@ -1,22 +1,167 @@
 [//]: # (title: Connect and retrieve data from databases)
 
-Simple guide how to start working data sources with Kotlin Notebooks.
+Kotlin Notebook offers capabilities for connecting to and retrieving data from various types of SQL databases, such as 
+MariaDB, PostgreSQL, MySQL, and SQLite. 
+Utilizing the [DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html), Kotlin Notebook can establish 
+connections to databases, execute SQL queries, and import the results for further operations.
+
+For detailed example, see the [Notebook in the KotlinDataFrame SQL Examples GitHub repository](https://github.com/zaleslaw/KotlinDataFrame-SQL-Examples/blob/master/notebooks/imdb.ipynb).
 
 ## Before you start
 
-* Download and install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
-* Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook).
-* Add a dependency to the DataFrame library (?)
+1. Download and install the latest version of [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/download/?section=mac).
+2. Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook) in IntelliJ IDEA.
+
+   > Alternatively, access the Kotlin Notebook plugin from **Settings/Preferences** | **Plugins** | **Marketplace** within IntelliJ IDEA.
+   >
+   {type="tip"}
+
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+5. Ensure you have access to an SQL database, such as MariaDB or MySQL.
+
+
+## Connect to database
+
+You can connect to and interact with an SQL database using specific functions from the [DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html). 
+You can use `DatabaseConfiguration` to establish a connection to your database and `getSchemaForAllSqlTables` to retrieve 
+the schema of all tables within it.
+
+Let's look at an example:
+
+1. **Import libraries:** Import the DataFrame library, which is essential for data manipulation tasks, along with
+the necessary Java libraries for SQL connectivity and utility functions: 
+
+```kotlin
+%use dataframe
```

Comment:
I see what you mean, but for this specific case I think it might be better to repeat this key information (importing DataFrame) helps with creating an end2end step list - people might arrive at the page searching at this topic missing the Before you start (and if they completed another guide before but they are opening a new notebook which is realistic they might altogether miss that step). It doesn't hurt if they repeat this single step so I think it should be fine.